tests/test_templates.py

```diff
@@ -128,11 +134,13 @@ def testComponent(self):
         tmplstr = "{run}_c_{component}_v{visit}"
         self.assertTemplate(tmplstr, "run2_c_output_v52", refMetricOutput)
 
-        tmplstr = "{run}_{component:?}_{visit}"
-        self.assertTemplate(tmplstr, "run2_52", refMetric)
-        self.assertTemplate(tmplstr, "run2_output_52", refMetricOutput)
-        self.assertTemplate(tmplstr, "run2_maskedimage.variance_52", refMaskedImage)
-        self.assertTemplate(tmplstr, "run2_output_52", refMetricOutput)
+        # We want this template to have both a directory and basename, to
+        # test that the right parts of the output are replaced.
+        tmplstr = "{run}_{component:?}/{visit}"
```

Comment:
I think if you changed this template to `"{component:?}/{run}_{component:?}_{visit}"` then you would test the directory and file part of the replacement without needing the extra test above with the filer.

Reply:
That's a good idea. I left the other test, since it's a little more obvious what's happening in it.